Source/Engine/Physics/Physics.h

```diff
@@ -514,11 +519,22 @@ DECLARE_SCRIPTING_TYPE_NO_SPAWN(Physics);
     /// <param name="dt">The delta time (in seconds).</param>
     API_FUNCTION() static void Simulate(float dt);
 
+    /// <summary>
+    /// Called during main engine loop to start physic simulation on all registered scenes.
+    /// </summary>
+    /// <param name="dt">The delta time (in seconds).</param>
+    static void SimulateAll(float dt);
+
     /// <summary>
     /// Called during main engine loop to collect physic simulation results and apply them as well as fire collision events.
     /// </summary>
     API_FUNCTION() static void CollectResults();
 
+    /// <summary>
+    /// Called during main engine loop to collect physic simulation results on all registered scenes and apply them as well as fire collision events.
+    /// </summary>
+    static void CollectResultsAll();
```

Comment:
Do we need separate `CollectResults` and `CollectResultsAll`? For me, it could by just `CollectResults` and collect results from all scenes by default.

Reply:
Yes, this is part of maintaining backwards compatibility. `CollectResults` will internally call `CollectResults` on the default scene and therefore maintains the ability of existing projects to manually collect results without needing to change their code.

 `CollectResultsAll` is called by the engine to collect results on all registered scenes. I think it's important to maintain this distinction.